src/chai/ManagedArray.inl

```diff
@@ -302,6 +306,9 @@ CHAI_HOST void ManagedArray<T>::reset()
 template<typename T>
 CHAI_INLINE
 CHAI_HOST_DEVICE size_t ManagedArray<T>::size() const {
+#if !defined(CHAI_DEVICE_COMPILE)
```

Comment:
I'm concerned about this change. Do you have a reproducer to show why this is necessary?

Reply:
This is probably not necessary for ManagedSharedPtr, this was necessary for the implementation of a ManagedVector like object but it can be necessary for a standar ManagedArray if you aren't careful.

```c++
using chai::ManageArray<double> = MA;
MA a1(10);
MA a2 = a1;

a1.reallocate(20);
assert(a1.data() == a2.data()) // True, we are looking at the same pointer record
assert(a1.size() == a2.size()); //This fails because the internal m_size member isn't updated for a2.
```

The 'size' needs to be held by the MA in order to be able to copy the metadata of the object to the device. However relying on the internal member can cause a disconnect between the allocation copies of a MA are trying to represent.